flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/SystemTypeInference.java

```diff
@@ -147,7 +159,9 @@ private static void checkScalarArgsOnly(List<StaticArgument> defaultArgs) {
         checkPassThroughColumns(declaredArgs);
 
         final List<StaticArgument> newStaticArgs = new ArrayList<>(declaredArgs);
-        newStaticArgs.addAll(PROCESS_TABLE_FUNCTION_SYSTEM_ARGS);
```

Comment:
I am curious what the system arguments mean. Is this something that the user needs to be aware of? I do not see this phrase in the Flip and there is no more information in the Jira.

Reply:
This in not entirely correct. A user-defined PTF can implement a TypeInference and avoid system args, but this is kind of second-level API.